src/data/nat/prime.lean

```diff
@@ -1004,6 +1004,17 @@ namespace nat
 
 theorem prime_three : prime 3 := by norm_num
 
+/--
+In most cases, we should not have global instances of `fact`; typeclass search only reads the head
+symbol and then tries any instances, which means that adding any such instance will cause slowdowns
+everywhere. We instead make them as lemmata and make them local instances as required.
+-/
+library_note "fact non-instances"
```

Comment:
Please move this (useful!) note to the definition of `fact`. That way in can be referenced everywhere where `fact` is available, and it's also visible quickly to people who "go-to definition" for `fact`.